src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -255,8 +256,22 @@ def get_path_to_port_config_file(hwsku=None, asic=None):
 
     # if 'hwsku.json' file is available, Check for 'platform.json' file presence,
     # if 'platform.json' is available, APPEND it. Otherwise, SKIP it.
+
+    """
+    This length check for interfaces in platform.json is performed to make sure
+    the cfggen does not fail if port configuration information is not present
+    TODO: once platform.json has all the necessary port config information
+          remove this check
+    """
+
     if os.path.isfile(hwsku_json_file):
-        port_config_candidates.append(os.path.join(platform_path, PLATFORM_JSON_FILE))
+        if os.path.isfile(os.path.join(platform_path, PLATFORM_JSON_FILE)):
+            json_file = os.path.join(platform_path, PLATFORM_JSON_FILE)
+            platform_data = json.loads(open(json_file).read())
+            interfaces = platform_data.get('interfaces', None)
+            if interfaces:
+                if len(interfaces) > 0:
```

Comment:
I believe `if interfaces:` should be enough, as it will return false if either `interfaces` is `None` or its length is 0. If you want to be more explicit, I recommend `if interfaces is not None and len(interfaces) > 0:`

Reply:
fixed